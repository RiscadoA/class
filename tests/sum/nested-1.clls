proc main() {
    cut {
        println("case x 1");
        case x of {
            |#A: println("close x 1"); close x
            |#B: println("case x 2"); case x of {
                |#A: println("wait x 1"); wait x; ()
                |#B: println("close x 2"); close x
            }
        }
        |x : choice of { |#A: wait |#B: choice of { |#A: close |#B: wait } }|
        println("#A x");
        #A x;
        println("wait x 2");
        wait x;
        ()
    }
};;
