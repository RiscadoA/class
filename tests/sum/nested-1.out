#A x
case x 1
close x 1
wait x 2
